### PR TITLE
Python Changes and Updates

### PR DESCRIPTION
**Pull request by erinxocon**
I have really wanted to use python with my matrix creator and the provided examples were only partially working, I'm assuming due to API changes and what not.  I've updated most of the examples and split up `test_sensors.py` to be more like their JS counterparts.  I also wrote a `utils.py` module that allows some easy setup for an event driven methodology.  If you like what I've done and want to include it don't merge it yet!  I'm still porting a couple of the examples and I want to add more comments.  I've also provided a `requirements.txt` and a Pipfile for so users can easily add the needed dependencies.  I'd also like to change the  `README.md` to relflect the changes I've made in the example files.

Let me know what you think!

### CODE REVIEW

**Comment by hpsaturn** [comment]
nice work! many thanks for your commits! we'll wait then!

**Comment by erinxocon** [comment]
@hpsaturn I've updated all the files I created/modify to try and comment line by line what is happening, feel free to correct anything!  I didn't touch the [README.rst](https://github.com/matrix-io/matrix-creator-malos/blob/master/src/python_test/README.md) as some of it differs after I rearranged some stuff to make it more pythonic.  I'll let you decide how you want to handle that.  If you want we can work in conjunction to create a new README. 

All and all I think it's ready to merge!

Cheers!

File: src/python_test/test_driver_info.py
Comment (hpsaturn) [inline src/python_test/test_driver_info.py]: Please set to localhost

**Review by hpsaturn — COMMENTED** [review]
Many thanks @erinxocon for your commits, all team right now are working on a big change on protosbuffers, I reviewed your examples and they all make sense and are fine, but the merge will take more time because of this change.

File: src/python_test/test_driver_info.py
Comment (erinxocon) [reply src/python_test/test_driver_info.py]: Oops will fix today!

**Comment by erinxocon** [comment]
@hpsaturn Good to hear!  I'd be happy to amend my commits with anything new that would be needed with the new protobuffers!

**Comment by erinxocon** [comment]
@hpsaturn I've merged in the new proto driver changes courtesy of @maciekrb!  Anything else to do?

**Comment by hpsaturn** [comment]
@erinxocon Good news, I add new reviewers and I think so we all ready.

**Review by maciekrb — APPROVED** [review]
Looks good to me, thank you very much @erinxocon. There is a minor tweak we have to add to make `zmq.setsockopt` compatible with python 2 and 3. I think we can do this after we merge.

**Pull request merged**
